internal/slack_integration/slack.go

```diff
@@ -77,26 +79,50 @@ func (b *Integration) handleEventAPI(ctx context.Context, event slackevents.Even
 		switch ev := event.InnerEvent.Data.(type) {
 		case *slackevents.MessageEvent:
 			if ev.ThreadTimeStamp == "" {
-				// Get channel info before notifying
-				channelInfo, err := b.client.Client.GetConversationInfo(&slack.GetConversationInfoInput{
-					ChannelID: ev.Channel,
-				})
-				if err != nil {
-					log.Printf("Error getting channel info: %v", err)
-					return
-				}
-
-				err = b.bot.Notify(ctx, ev.Channel, channelInfo.Name)
+				// First, notify immediately with just the channel ID
+				err := b.bot.Notify(ctx, ev.Channel)
 				if err != nil {
 					log.Printf("Error notifying: %v", err)
 				}
+
+				// Then, trigger background channel info fetch if needed
+				go b.ensureChannelInfo(ctx, ev.Channel)
```

Comment:
@rajatgoel doing this way, after writing this feeling like I should just do a river job for this, otherwise its not guaranteed to fill correctly.